examples/tcp-close/client.rs

```diff
@@ -150,4 +151,124 @@ impl TcpClient {
+                    let pop_qr: demi_qresult_t = self.libos.wait(pop_qt, None)?;
+
+                    match pop_qr.qr_opcode {
+                        demi_opcode_t::DEMI_OPC_POP => {
+                            let sga: demi_sgarray_t = unsafe { pop_qr.qr_value.sga };
+                            let received_len: u32 = sga.sga_segs[0].sgaseg_len;
+                            self.libos.sgafree(sga)?;
+                            // 0 len pop represents socket closed from other side.
+                            assert_eq!(
+                                received_len, 0,
+                                "server should have had closed the connection, but it has not"
+                            );
+                            println!("server disconnected (pop returned 0 len buffer)");
+                        },
+                        demi_opcode_t::DEMI_OPC_FAILED => {
+                            let errno: i32 = qr.qr_ret;
+                            assert_eq!(
+                                errno,
+                                libc::ECONNRESET,
+                                "server should have had closed the connection, but it has not"
+                            );
+                            println!("server disconnected (ECONNRESET)");
+                        },
+                        qr_opcode => panic!("unexpected result (qr_opcode={:?})", qr_opcode),
+                    }
+                },
+                demi_opcode_t::DEMI_OPC_FAILED => panic!("operation failed (qr_ret={:?})", qr.qr_ret),
+                qr_opcode => panic!("unexpected result (qr_opcode={:?})", qr_opcode),
+            }
+
+            // Close TCP socket.
+            self.libos.close(sockqd)?;
+        }
+
+        Ok(())
+    }
+
+    /// Attempts to make several connections concurrently.
+    pub fn run_concurrent_expecting_server_to_close_sockets(&mut self, num_clients: usize) -> Result<()> {
+        let mut q_descs: Vec<QDesc> = Vec::default();
+        let mut q_tokens: Vec<QToken> = Vec::default();
+
+        // Create several TCP sockets and connect.
+        for _i in 0..num_clients {
+            let qd: QDesc = self.libos.socket(AF_INET, SOCK_STREAM, 0)?;
+            q_descs.push(qd);
+            let qt: QToken = self.libos.connect(qd, self.remote)?;
+            q_tokens.push(qt);
+        }
+
+        // Wait for all connections to be established and then closed by the server.
+        loop {
+            if self.clients_closed == num_clients {
+                // Stop when enough connections were closed.
+                break;
+            }
+
+            let qr: demi_qresult_t = {
+                let (index, qr): (usize, demi_qresult_t) = self.libos.wait_any(&q_tokens, None)?;
+                let _qt: QToken = q_tokens.remove(index);
+                qr
+            };
+
+            match qr.qr_opcode {
+                demi_opcode_t::DEMI_OPC_CONNECT => {
+                    let sock_qd: QDesc = qr.qr_qd.into();
```

Comment:
Our code base use `sockqd` for naming this variable. Could you change your code to match the existing pattern?